### PR TITLE
Better README#3

### PR DESCRIPTION
**Pull request by arashabzd**
Added a short usage section with a diagram of the API to README.md and two simple examples that show the general workflow of defining and training a conditional/unconditional flow.

### CODE REVIEW

**Comment by arturbekasov** [comment]
Thanks for taking the time to write this up. I think this is a good start for better documentation.

Before I can merge this, a few comments:
1. While useful, I think the diagram is somehwat too technical for a readme. I propose dropping it for now.
1. On other changes in the readme: I think most of the changes could be replaced with a short snippet of code of the form `flow = Flow(transform, base_distribution)`, showing an example of how to create a `transform` and a `base_distribution`, and only then point to `nflows.transformations` and `nflows.distributions` as places where to find more transforms and distributions, and to `examples` for more examples.
1. In the notebooks: sublassing the `Flow` class is a rather convoluted way to create a flow. It's much easier and much more concise to create a `Flow` instance with the desired transformation and base distribution (see 2.). Could this be updated?

**Comment by arashabzd** [comment]
Yes, sure. 
Notebooks got updated.
About the code snippet, I'm not sure which specific transform to use since each transform has different parameters. Do you have any preferences?

**Comment by arturbekasov** [comment]
No need to get into transform parameters. This is what I have in mind:

To define a flow:
```python
from nflows import transforms, distributions, flows

# Define an invertible transformation.
transform = transforms.CompositeTranform([
    transforms.AffineCouplingTransform(...),
    transforms.RandomPermutation(...)
])

# Define a base distribution.
base_distribution = distributions.StandardNormal(...)

# Combine into a flow.
flow = flows.Flow(transform=transform, distribution=base_distribution)
```

To evaluate log probabilities of inputs:
```python
log_prob = flow.log_prob(inputs)
```

To sample from flow:
```python
samples = flow.sample(num_samples=...)
```

File: README.md [hunk 1]
```diff
@@ -14,6 +14,37 @@ You can install all the dependencies using the `environment.yml` file to create
...
+```python
+log_prob = flow.log_prob(inputs)
+```
+
+To sample from the flow:
+```python
+samples = flow.sample(num_samples)
+```
+
+Additional examples of the workflow are provided in [examples folder](https://github.com/arashabzd/nflows/tree/better-readme%233/examples).
```
Comment (arturbekasov) [inline README.md:47]: This should point to `https://github.com/bayesiains/nflows/tree/master/examples`. Or even simply `examples` if relative links work on GitHub?

**Comment by arturbekasov** [comment]
Yep, this looks good. See the last nitpick: happy to merge after it's fixed.

Any further improvements could be done outside of this PR.

Thanks!

**Comment by imurray** [comment]
@arashabzd thanks so much for contributing these docs and examples! It's appreciated. (And thanks @arturbekasov for shepherding the PR.)

**Pull request merged**